src/common/src/util/ordered/serde.rs

```diff
@@ -154,14 +156,39 @@ impl OrderedRowDeserializer {
     pub fn get_order_types(&self) -> &[OrderType] {
         &self.order_types
     }
+
+    pub fn deserialize_prefix_len_with_column_indices(
+        &self,
+        key: &[u8],
+        column_indices: Vec<usize>,
+    ) -> memcomparable::Result<usize> {
+        use crate::types::ScalarImpl;
+        let mut len: usize = 0;
+        for index in column_indices {
+            let data_type = &self.data_types[index];
+            let order_type = &self.order_types[index];
+            let data = &key[len..];
+            let mut deserializer = memcomparable::Deserializer::new(data);
+            deserializer.set_reverse(*order_type == OrderType::Descending);
+
+            let _data_size = ScalarImpl::encoding_data_size(data_type, &mut deserializer)?;
+            len += deserializer.position();
```

Comment:
So this function only calculate the bytes length?

Reply:
yes, it only return the prefix_len to caller and avoid mem copy